src/patterns.rs

```diff
@@ -97,3 +173,14 @@ impl Rewrite for Pat {
         }
     }
 }
+
+impl Rewrite for FieldPat {
+    fn rewrite(&self, context: &RewriteContext, width: usize, offset: Indent) -> Option<String> {
+        let pat = self.pat.rewrite(context, width, offset);
+        if self.is_shorthand {
+            pat
+        } else {
+            Some(format!("{}: {}", self.ident.to_string(), try_opt!(pat)))
```

Comment:
I'd wrap this in a `wrap_str` to be sure, or do a separate rewrite of the pat with a budget of `width - ident.to_string().len() - 2`.